lib/tasks/swagger_generate.rake

```diff
@@ -295,6 +358,18 @@ class SwaggerGenerator
   end
 end
 
+GENERATOR_BLACKLIST_ATTRIBUTES           = [
+  :resource_timestamp, :resource_timestamps, :resource_timestamps_max, :tenant_id
+].to_set.freeze
+GENERATOR_ALLOW_BLACKLISTED_ATTRIBUTES   = {
+  :tenant_id => ['Source', 'Endpoint', 'Authentication'].to_set.freeze
+}
+GENERATOR_WHITELIST_PROVIDER_DEFINITIONS = [
```

Comment:
Wonder if we could find a better name for this?  `GENERATOR_WHITELIST_PROVIDER_DEFINITIONS` sounds to me like we allow these things and default drop everything else.

It looks like it really is just a list of type that cannot be created through this API (aka populated by refresh).  How about `GENERATOR_READ_ONLY_DEFINITIONS` ?